### PR TITLE
HHH-17635 - register custom user types

### PR DESCRIPTION
**Pull request by jrenaat**
when contributed through a service provider

https://hibernate.atlassian.net/browse/HHH-17635

### CODE REVIEW

**Comment by beikov** [comment]
Please also add a test for this.

File: hibernate-core/src/test/resources/META-INF/services/org.hibernate.boot.model.TypeContributor [hunk 1]
```diff
@@ -0,0 +1 @@
+org.hibernate.orm.test.type.contributor.usertype.ServiceLoadedCustomUserTypeTypeContributor
```
Comment (beikov) [inline hibernate-core/src/test/resources/META-INF/services/org.hibernate.boot.model.TypeContributor:1]: Please use `@ServiceRegistry(javaServices = @JavaService(role = TypeContributor.class, impls = ServiceLoadedCustomUserTypeTypeContributor.class))` on the particular test instead.

File: hibernate-core/src/test/java/org/hibernate/orm/test/service/ClassLoaderServiceImplTest.java [hunk 2]
```diff
@@ -96,7 +94,12 @@ public void testStoppableClassLoaderService() {
 
 	private TypeContributor getTypeContributorServices(ServiceRegistry serviceRegistry) {
 		final ClassLoaderService classLoaderService = serviceRegistry.getService( ClassLoaderService.class );
-		final Collection<TypeContributor> typeContributors = classLoaderService.loadJavaServices( TypeContributor.class );
+		final Collection<TypeContributor> typeContributors =
+				new ArrayList<>( classLoaderService.loadJavaServices( TypeContributor.class ) );
+		typeContributors.removeIf( typeContributor ->
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/service/ClassLoaderServiceImplTest.java:99]: Please revert the change

**Review by mbladel — APPROVED** [review]
LGTM
